gradle/java.gradle

```diff
@@ -16,109 +16,104 @@
  */
 import org.apache.geode.gradle.plugins.DependencyConstraints
 
-allprojects {
-  // There exist peculiarities surrounding our handling of ClassLoaders and cross-module resources,
-  // causing test failures if we apply the newer 'java-library' plugin.
-  if (project.name.endsWith("geode-all-bom")) {
-    // This anti-pattern is a workaround -- java-platform must be applied before java or java-library
-    // to avoid conflicts over redefining certain configurations.
-    // A better solution, per GEODE-6383, would be to refactor this file's configuration to be
-    // "opt in", rather than having the root project inject this configuration into all projects.
-    apply plugin: 'java-platform'
-  }
-  apply plugin: 'java-library'
-
-  sourceCompatibility = 1.8
-  targetCompatibility = 1.8
-  compileJava.options.encoding = 'UTF-8'
+// There exist peculiarities surrounding our handling of ClassLoaders and cross-module resources,
+// causing test failures if we apply the newer 'java-library' plugin.
+if (project.name.endsWith("geode-all-bom")) {
+  // This anti-pattern is a workaround -- java-platform must be applied before java or java-library
+  // to avoid conflicts over redefining certain configurations.
+  // A better solution, per GEODE-6383, would be to refactor this file's configuration to be
+  // "opt in", rather than having the root project inject this configuration into all projects.
```

Comment:
Comment refers to this issue being fixed, but using a different GEODE ticket

Reply:
Hmmm.  Nothing quite like semantics...

I had originally opened GEODE-6383 as an overarching "parent" ticket for multiple sub-tickets to be attached to, with the driving effort of restoring modularity.  But you're absolutely right that this PR does it in one fell swoop.  The remaining work is to examine which projects should actually have a particular plugin applied to them, but that is beyond the scope of "restore modularity" per 6383.

I'll fix the comment and update the JIRA, since yes, this really is resolving 6383 in its entirety.

Reply:
Hmmm. Nothing quite like semantics...

I had originally opened GEODE-6383 as an overarching "parent" ticket for multiple sub-tickets to be attached to, with the driving effort of restoring modularity. But you're absolutely right that this PR does it in one fell swoop. The remaining work is to examine which projects should actually have a particular plugin applied to them, but that is beyond the scope of "restore modularity" per 6383.

I'll fix the comment and update the JIRA, since yes, this really is resolving 6383 in its entirety.

Reply:
I can't entirely decide of it's better to refer to GEODE-6611, or if that line should be prune in its entirety.